Output PURE calls

itsMapleLeaf opened the pull request:
Closes #6

itsMapleLeaf committed:
install @babel/helper-annotate-as-pure

itsMapleLeaf committed:
annotate tw calls as pure

itsMapleLeaf committed:
Merge branch 'main' into pure-calls

sastan reviewed (approved):
LGTM

itsMapleLeaf commented on test/fixtures/tw-import/output.js:
```diff
@@ -1,3 +1,3 @@
 import { tw } from 'twind'
 const buttonClass = tw`bg-blue-500`
```
Actually, after a second look, this call should be annotated too, shouldn't it?

sastan replied:
I didn't look at all code only at the changes. Sorry

itsMapleLeaf committed:
update test scripts

- run locally, use --watchAll
- add a script for running all tests in ci

itsMapleLeaf committed:
import locally under a new name

since we now look at macro references and do something with them,
this is a good step to ensure the macro references are disambiguated
from the actual `tw` function

itsMapleLeaf committed:
rename macro references and annotate them

sastan commented on src/macro.ts:
```diff
@@ -1,8 +1,11 @@
 import { findJsxAttributeByName, getJsxAttributeName, getJsxAttributeValue } from './jsx-attribute'
 
-function twindMacro({ state }: MacroParams) {
+const localImportName = '_macro_tw'
```
I think babel as a way to generate a local name that does not interfere with any existing name. It's a while ago ... I can not remember the function.

itsMapleLeaf replied:
Is it this? https://babeljs.io/docs/en/babel-helper-module-imports

sastan replied:
Looks like it. 👍

itsMapleLeaf replied:
Ah, I did a little digging (a.k.a stealing more ideas from twin.macro 😆) and it looks like this is a part of babel's core API: `program.scope.generateUid('tw')`

itsMapleLeaf commented:
Changes here are done. If it looks good, we can merge and release

sastan commented:
I have added you to @twind/macro publishers on npm. You should be able to publish it using `yarn publish`. This script uses [np](https://www.npmjs.com/package/np) which really helps with publishing packages.

itsMapleLeaf committed:
use babel's API to generate a unique name in scope

itsMapleLeaf committed:
add test case for using macro and twind in same file

plus a renamed macro reference

itsMapleLeaf committed:
add test case for extra props

sastan commented:
LGTM
